### PR TITLE
[SPARK-22448][ML] Added sum function to Summerizer and MultivariateOn…

### PR DESCRIPTION
**Pull request by dedunumax**
[SPARK-22448][ML] Added sum function to Summerizer and MultivariateOnlineSummarizer

## What changes were proposed in this pull request?

This change adds sum() to Summerizer and MultivariateOnlineSummarizer.

## How was this patch tested?

Added unit test to make sure it works. Manually tested in spark-shell too.

### CODE REVIEW

**Comment by dedunumax** [comment]
cc @rxin @cloud-fan @gatorsmile

**Comment by dbtsai** [comment]
ok to test

**Comment by dbtsai** [comment]
Why do you want to add this? Once we have mean, it's easy to compute sum.

**Comment by dedunumax** [comment]
Having sum as a basic statistic will make the API user-friendly. I'm thinking about implementing other functions as well. Do you think it is not worth to implement this?

**Comment by WeichenXu123** [comment]
I doubt that this will slow down the summarizer performance because you add sum statistics internally (and this sum value will possible to overflow). 
We can directly use `count * mean` to get the sum if we want to use it.

**Comment by dedunumax** [comment]
I see, I will change the code like that.

**Comment by dedunumax** [comment]
cc @WeichenXu123 @dbtsai

**Comment by dedunumax** [comment]
cc @WeichenXu123 @dbtsai

File: mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala [hunk 1]
```diff
@@ -562,6 +573,23 @@ private[ml] object SummaryBuilderImpl extends Logging {
...
+    def sum: Vector = {
+      require(requestedMetrics.contains(Sum))
+      require(totalWeightSum > 0, s"Nothing has been added to this summarizer.")
+
+      val realSum = Array.ofDim[Double](n)
+      var i = 0
+      val len = currMean.length
+      while (i < len) {
+        realSum(i) = currMean(i) * weightSum(i)
+        i += 1
```
Comment (mahmoudmahdi24) [inline mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala]: Please avoid using mutable values, use foldLeft for example to solve this.

Comment (dedunumax) [reply mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala]: I will change that.

**Pull request closed without merging**